lms/djangoapps/grades/signals/handlers.py

```diff
@@ -144,6 +145,8 @@ def score_published_handler(sender, block, user, raw_earned, raw_possible, only_
             modified=score_modified_time,
             score_db_table=ScoreDatabaseTableEnum.courseware_student_module,
         )
+        if isinstance(block, ScorableXBlockMixin):
+            block.set_score(Score(raw_earned=raw_earned, raw_possible=raw_possible))
```

Comment:
Although we may not need it at this time, calling `set_score`  before the `PROBLEM_RAW_SCORE_CHANGED` is signaled makes the code easier to understand - since it's updated before grades are updated.